src/entity/Living.php

```diff
@@ -293,10 +293,26 @@ public function saveNBT() : CompoundTag{
 		return $nbt;
 	}
 
+	/**
+	 * Returns whether this entity has not block line of sight to another entity.
+	 *
+	 * @throws \InvalidArgumentException if the entities are not in the same world
+	 */
 	public function hasLineOfSight(Entity $entity) : bool{
-		//TODO: head height
+		$world = $this->getWorld();
+		if ($world !== $entity->getWorld()) {
+			throw new \InvalidArgumentException("Target entity is not in the same world");
+		}
+		$start = $this->getEyePos();
+		$end = $entity->getEyePos();
+		if ($end->subtractVector($start)->normalize()->lengthSquared() > 0) {
```

Comment:
what is this supposed to do? the modulus of a normalized vector is always (approximately) 1. did you want to check if they are the same point instead?